sql/core/src/test/resources/sql-tests/results/udf/postgreSQL/udf-aggregates_part1.sql.out

```diff
@@ -109,7 +109,7 @@ struct<udf(var_samp(b)):double>
 -- !query
 SELECT udf(stddev_pop(CAST(b AS Decimal(38,0)))) FROM aggtest
 -- !query schema
-struct<udf(stddev_pop(CAST(b AS DECIMAL(38,0)))):double>
+struct<udf(stddev_pop(ansi_cast(b as decimal(38,0)))):double>
```

Comment:
Why do we need this change?

Reply:
The difference is introduced by `NonSQLExpression.sql` which use `toString` instead of `sql` and `Cast` has different implementation between `toString` and `sql`. Then the change happened.  Is this get your point ?

Reply:
Why do we couple this with `NonSQLExpression`? Can we change `AnsiCast.sql` directly?

Reply:
Not couple them, just follow the semantics of `NonSQLExpression` to make child toString instead of sql. `AnsiCast` could be seen as an another issue.

You mean we could override `CastBase.sql` to print `ansi_cast` like what `CastBase.toString` do ? Seems we have not provided a public api to user with `AnsiCast`. I wonder if `AnsiCast` is only used in Spark internal and not exposed.